lib/rom/sql/extensions/postgres/types/array_types.rb

```diff
@@ -0,0 +1,72 @@
+require 'rom/sql/type_extensions'
+
+module ROM
+  module SQL
+    module Postgres
+      module Types
+        # @api private
+        class ArrayTypes
+          attr_reader :elements
+
+          attr_reader :constructor
+
+          attr_reader :base_write_type
+
+          attr_reader :base_read_type
+
+          def initialize(base_write_type, base_read_type)
+            @elements = {}
+            @base_write_type = base_write_type
+            @base_read_type = base_read_type
+            @constructor = proc { |db_type, member|
+              -> arr {
+                if member
```

Comment:
this should be
```ruby
if member
  -> arr {  }
else
  -> arr {  }
end
```
i.e. the condition can be checked only once per type rather than once per cast

Reply:
@flash-gordon ohh nice catch! I'll address this.